compiler/x/codegen/X86Ops.hpp

```diff
@@ -414,6 +423,11 @@ class TR_X86OpCode
       // OpCodes beyond FENCE are not X86 instructions and hence they do not have binary encodings.
       return op < FENCE ? _binaries[op].encode<TBuffer>(cursor, rex) : cursor;
       }
+   static inline void finalize(TR_X86OpCodes op, uint8_t* cursor)
+      {
+      if (op < FENCE)
```

Comment:
_properties would not work since the op can encoded as either SSE or AVX instructions; only finalization only works on AVX instructions. Adding such property does not help as we still have to detect if the instruction is SSE or AVX. An instruction is AVX if and only if its first byte is 0xC4 or 0xC5.
The root cause of this issue is that AVX instructions have three operands but SSE instructions only have two. Our CG defines instructions with two-operand as well. We need somehow convert the instruction to three-operand mode; but required information is only available after operands are encoded. I am working on a better design for such conversion but have not finalized yet.
This approach is one required least code refactoring and hence I would like to deliver as a intermediate path to unblock queued features that require AVX. As mentioned in the X86 Binary Encoding and SIMD changesets, we are working on supporting three-operand instructions. Once three-operand instructions are properly supported, this "finalization" approach will be replaced.